pkg/specgen/generate/oci_freebsd.go

```diff
@@ -18,11 +19,15 @@ import (
 
 // SpecGenToOCI returns the base configuration for the container.
 func SpecGenToOCI(ctx context.Context, s *specgen.SpecGenerator, rt *libpod.Runtime, rtc *config.Config, newImage *libimage.Image, mounts []spec.Mount, pod *libpod.Pod, finalCmd []string, compatibleOptions *libpod.InfraInherit) (*spec.Spec, error) {
-	if s.ImageOS != "freebsd" && s.ImageOS != "linux" {
-		return nil, fmt.Errorf("unsupported image OS: %s", s.ImageOS)
+	imageOS := s.ImageOS
+	if imageOS == "" {
+		imageOS = runtime.GOOS
```

Comment:
Shouldn't this validation and assignment be on the front-end instead of specgen.

Reply:
That would probably be a better approach - I'll re-work this tomorrow.

Reply:
I moved the logic to specgenutil.FillOutSpecGen so that it can be shared between create and run.